cadcdata/cadcdata/storageinv.py

```diff
@@ -0,0 +1,826 @@
+
+    args = parser.parse_args()
+    subject = net.Subject.from_cmd_line_args(args)
+    client = StorageInventoryClient(subject, args.service, host=args.host,
+                                    insecure=args.insecure)
+    for id in args.identifier:
+        try:
+            file_info = execute_cmd(args, client.cadcinfo, {'id': id})
+            print('CADC Storage Inventory identifier {}:'.format(id))
+            print('\t {:>15}: {}'.format('name', file_info.name))
+            print('\t {:>15}: {}'.format('size', file_info.size))
+            print('\t {:>15}: {}'.format('type', file_info.file_type))
+            print('\t {:>15}: {}'.format('encoding', file_info.encoding))
+            print('\t {:>15}: {}'.format('last modified',
+                                         date2ivoa(file_info.lastmod)))
+            print('\t {:>15}: {}'.format('md5sum', file_info.md5sum))
+        except exceptions.NotFoundException:
+            logger.error('Identifier {} not found in the CADC Storage '
+                         'Inventory'.format(id), exit_after=False)
+            continue
+
+
+def cadcremove_cli():
+    parser = util.get_base_parser(subparsers=False,
+                                  version=version.version,
+                                  service=DEFAULT_RESOURCE_ID,
+                                  auth_required=True)
+    parser.description = (
+        'Remove files from the CADC Storage Inventory')
+
+    parser.add_argument(
+        'identifier',
+        help='unique identifier (URI) given to the file in the CADC '
+             'Storage Inventory', nargs='+')
+    parser.epilog = (
+        'Examples:\n'
+        '- Use certificate to remove a file from the storage inventory:\n'
+        '       cadcremove --cert ~/.ssl/cadcproxy.pem cadc:CFHT/700000o.fz\n')
+
+    args = parser.parse_args()
+    subject = net.Subject.from_cmd_line_args(args)
+    client = StorageInventoryClient(subject, args.service, host=args.host,
+                                    insecure=args.insecure)
+    for id in args.identifier:
+        execute_cmd(args, client.cadcremove, {'id': id})
+
+
+def execute_cmd(args, cmd, cmd_args):
+    # handle errors
+    errors = [0]
+
+    def handle_error(msg, exit_after=True):
+        """
+        Prints error message and exit (by default)
+        :param msg: error message to print
+        :param exit_after: True if log error message and exit,
+        False if log error message and return
+        :return:
+        """
+
+        errors[0] += 1
+        logger.error(msg)
+        if exit_after:
+            sys.exit(-1)  # TODO use different error codes?
+    if args.verbose:
+        logging.basicConfig(level=logging.INFO, stream=sys.stdout)
+    elif args.debug:
+        logging.basicConfig(level=logging.DEBUG, stream=sys.stdout)
+    else:
+        logging.basicConfig(level=logging.WARN, stream=sys.stdout)
+
+    try:
+        return cmd(**cmd_args)
+    except exceptions.UnauthorizedException:
```

Comment:
I am unsure of the logic by which the specific exceptions are handled here, but should the AttributeError be on the special list, due to the `replace` parameter in the `cadcput` call?